lambdas/services/bulk_upload_service.py

```diff
@@ -3,61 +3,78 @@
+)
+from repositories.bulk_upload.bulk_upload_s3_repository import BulkUploadS3Repository
+from repositories.bulk_upload.bulk_upload_sqs_repository import BulkUploadSqsRepository
+from utils import lloyd_george_validator
 from utils.audit_logging_setup import LoggingService
 from utils.exceptions import (
     DocumentInfectedException,
     InvalidMessageException,
     PatientRecordAlreadyExistException,
     PdsTooManyRequestsException,
     S3FileNotFoundException,
-    TagNotFoundException,
     VirusScanFailedException,
     VirusScanNoResultException,
 )
-from utils.lloyd_george_validator import LGInvalidFilesException, validate_lg_file_names
+from utils.lloyd_george_validator import LGInvalidFilesException
 from utils.request_context import request_context
 from utils.unicode_utils import (
     contains_accent_char,
     convert_to_nfc_form,
     convert_to_nfd_form,
 )
-from utils.utilities import create_reference_id
 
 logger = LoggingService(__name__)
 
 
 class BulkUploadService:
-    def __init__(
-        self,
-    ):
-        self.s3_service = S3Service()
-        self.dynamo_service = DynamoDBService()
-        self.sqs_service = SQSService()
-
-        self.staging_bucket_name = os.environ["STAGING_STORE_BUCKET_NAME"]
-        self.lg_bucket_name = os.environ["LLOYD_GEORGE_BUCKET_NAME"]
-        self.lg_dynamo_table = os.environ["LLOYD_GEORGE_DYNAMODB_NAME"]
-        self.bulk_upload_report_dynamo_table = os.environ["BULK_UPLOAD_DYNAMODB_NAME"]
-        self.invalid_queue_url = os.environ["INVALID_SQS_QUEUE_URL"]
-        self.metadata_queue_url = os.environ["METADATA_SQS_QUEUE_URL"]
+    def __init__(self):
+        self.dynamo_repository = BulkUploadDynamoRepository()
+        self.sqs_repository = BulkUploadSqsRepository()
+        self.s3_repository = BulkUploadS3Repository()
+
         self.pdf_content_type = "application/pdf"
 
-        self.dynamo_records_in_transaction: list[NHSDocumentReference] = []
-        self.source_bucket_files_in_transaction = []
-        self.dest_bucket_files_in_transaction = []
         self.file_path_cache = {}
 
+    def process_message_queue(self, records: list):
+        for index, message in enumerate(records, start=1):
+            try:
+                logger.info(f"Processing message {index} of {len(records)}")
+                self.handle_sqs_message(message)
+            except PdsTooManyRequestsException as error:
+                logger.error(error)
+                logger.info("Cannot process for now due to PDS rate limit reached.")
+                logger.info(
+                    "All remaining messages in this batch will be returned to sqs queue to retry later."
+                )
+
+                all_unprocessed_message = records[index - 1 :]
+                for unprocessed_message in all_unprocessed_message:
+                    self.sqs_repository.put_sqs_message_back_to_queue(
+                        unprocessed_message
+                    )
+                return
```

Comment:
would it be good if we raise a BulkUploadException here instead of the return statement?